doc/source/ray-air/examples/batch_tuning.ipynb

```diff
@@ -0,0 +1,1197 @@
+    "print(f\"s3_files: {s3_files}\")\n",
+    "print(f\"Locations: {sample_locations}\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0bf5b850-fa35-4397-bea6-b8c21fd754cd",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Function to read a pyarrow.Table object using pyarrow parquet \n",
+    "def read_data(file: str, sample_id: np.int32) -> pd.DataFrame:\n",
+    "    \n",
+    "    df = pq.read_table(\n",
+    "        file,\n",
+    "        filters=[\n",
+    "            (\"passenger_count\", \">\", 0),\n",
+    "            (\"trip_distance\", \">\", 0),\n",
+    "            (\"fare_amount\", \">\", 0),\n",
+    "            (\"pickup_location_id\", \"in\", [264, 265]),\n",
+    "            (\"dropoff_location_id\", \"not in\", [264, 265]), \n",
+    "            (\"dropoff_location_id\", \"=\", sample_id)\n",
+    "        ],\n",
+    "        columns=[\n",
+    "            \"pickup_at\",\n",
+    "            \"dropoff_at\",\n",
+    "            \"pickup_location_id\",\n",
+    "            \"dropoff_location_id\",\n",
+    "            \"passenger_count\",\n",
+    "            \"trip_distance\",\n",
+    "            \"fare_amount\",\n",
+    "        ],\n",
+    "    ).to_pandas()\n",
+    "\n",
+    "    return df\n",
+    "\n",
+    "# Function to transform a pandas dataframe\n",
+    "def transform_df(the_df: pd.DataFrame) -> pd.DataFrame:\n",
+    "    df = the_df.copy()\n",
+    "    \n",
+    "    df[\"trip_duration\"] = (df[\"dropoff_at\"] - df[\"pickup_at\"]).dt.seconds\n",
+    "    df = df[df[\"trip_duration\"] > 60]\n",
+    "    df = df[df[\"trip_duration\"] < 24 * 60 * 60] \n",
+    "    df.drop([\"dropoff_at\", \"pickup_at\", \"pickup_location_id\", \"fare_amount\"]\n",
+    "            , axis=1, inplace=True)\n",
+    "    df[\"dropoff_location_id\"] = df[\"dropoff_location_id\"].fillna(-1)\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "72369cbf-e346-441c-b7e3-9280ccabc7f6",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Define your Ray Tune Search Space and Search Algorithm <a class=\"anchor\" id=\"define_search_space\"></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e310b13e-ba83-4c47-84e6-d1e916483b2b",
+   "metadata": {},
+   "source": [
+    "**First, define a search space of experiment to run.** \n",
+    "> The search space together with a search algorithm determine which experiments will be run.  \n",
+    "\n",
+    "Common search algorithms include grid search, random search, Bayesian search, Hyperopt, and Optuna.  For more details, see [Working with Tune Search Spaces](https://docs.ray.io/en/master/tune/tutorials/tune-search-spaces.html#tune-search-space-tutorial).  Deciding the best combination of search space and search algorithm is part of the art of being a Data Scientist and depends on the data, algorithm, and problem being solved!  \n",
+    "\n",
```

Comment:
let's explain once again here that we are not running a hyperparam tuning job, we are just defining our search space in a way to allow for batch tuning to be conducted (multiple partitions and multiple models per partition)